### ISSUE TITLE
qs-title escape() breaks Reddit sharing for most unicode characters

### ISSUE THREAD

**Issue by mjones-feral** [Upstatement/quickshare#22]
Strings shared to Reddit are escaped but not encoded with encodeURIComponent. This is incorrect and causes most unicode characters to appear still encoded in the Reddit post title. This is a major problem with Russian localisation.

This was previously an issue with Twitter and has since been fixed but a similar issue still persists for Reddit. 

I have submitted a pull request with a fix for this problem very similar to the previous fix for Twitter that is now part of Master. https://github.com/Upstatement/quickshare/pull/21

I would really appreciate it if this was merged into Master (or some feedback was given). https://github.com/Upstatement/quickshare/pull/21

Thanks for this great tool,
Michael

https://github.com/Upstatement/quickshare/pull/21

**Comment by emmagard** [comment]
@mjones-feral Thanks for finding this issue and submitting a pull request. I will take care of making corrections sometime this week.

**Comment by mjones-feral** [comment]
@emmagard Thank you, much appreciated.

**Comment by emmagard** [comment]
@mjones-feral I push up a version to the master quickshare repo which addresses your issue (see PR #22). I removed the escape() and unescape() function from the repo and replaced it with encodeURIComponent() or decodeURIComponent(). 

Instead of adding encodeURIComponent() to the extractParams function in reddit.js, I added it to the _defaultParams function in utilities.js. This revision seems to fix the title encoding issue with Reddit. Feel free to file another issue if you are still having problems.

Thanks again for pointing out this issue.
